src/EnergyPlus/UnitarySystem.cc

```diff
@@ -11776,6 +11802,45 @@ namespace UnitarySystems {
         //  UnitarySystem(UnitarySysNum)%SuppHeatPartLoadFrac = PartLoadRatio
     }
 
+    void UnitarySys::setEMSSuppCoilStagePLR(EnergyPlusData &state)
+    {
+        bool useMaxedSpeed = false;
+        int SpeedNumEMS = ceil(this->m_EMSOverrideSuppCoilSpeedNumValue);
+        if (SpeedNumEMS > this->m_NumOfSpeedSuppHeating) {
+            SpeedNumEMS = this->m_NumOfSpeedSuppHeating;
+            useMaxedSpeed = true;
+        }
+        this->m_SuppHeatingSpeedNum = SpeedNumEMS;
+        if (useMaxedSpeed) {
+            this->m_CoilSpeedErrIdx++;
+            ShowRecurringWarningErrorAtEnd(state,
+                                           format("Wrong coil speed EMS override value, for unit=\"{}\". Exceeding maximum coil speed "
+                                                  "level. Speed level is set to the maximum coil speed level allowed.",
+                                                  this->m_SuppHeatCoilName),
+                                           this->m_CoilSpeedErrIdx,
+                                           this->m_EMSOverrideSuppCoilSpeedNumValue,
+                                           this->m_EMSOverrideSuppCoilSpeedNumValue,
+                                           _,
+                                           "",
+                                           "");
+        }
+        if (this->m_SuppHeatingSpeedNum == 1) {
+            this->m_SuppHeatingSpeedRatio = 0.0;
+            this->m_SuppHeatingCycRatio = this->m_EMSOverrideSuppCoilSpeedNumValue - floor(this->m_EMSOverrideSuppCoilSpeedNumValue);
+            if (useMaxedSpeed || this->m_SuppHeatingCycRatio == 0) {
+                this->m_SuppHeatingCycRatio = 1;
+            }
+            this->m_SuppHeatPartLoadFrac = this->m_SuppHeatingCycRatio;
+        } else {
+            this->m_SuppHeatingCycRatio = 1.0;
```

Comment:
I found a bug during my testing that if the SpeedNumEMS is 0.0 then it went to this condition and set the m_SuppHeatingCycRatio and m_SuppHeatingSpeedRatio to be 1.0, which is not correct, they should be 0.0.

Reply:
When should SpeedNum = 0? when the coil is off? Having a SpeedNum, CycRatio, and SpeedRatio is tricky.